File: chazm-model/src/main/java/runtimemodels/chazm/model/Entities.java
package runtimemodels.chazm.model;

import runtimemodels.chazm.api.entity.*;
import runtimemodels.chazm.api.id.UniqueId;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

class Entities {

    final Map<UniqueId<Agent>, Agent> agents = new ConcurrentHashMap<>();
    final Map<UniqueId<Attribute>, Attribute> attributes = new ConcurrentHashMap<>();
    final Map<UniqueId<Capability>, Capability> capabilities = new ConcurrentHashMap<>();
    final Map<UniqueId<Characteristic>, Characteristic> characteristics = new ConcurrentHashMap<>();
    final Map<UniqueId<InstanceGoal>, InstanceGoal> instanceGoals = new ConcurrentHashMap<>();
    final Map<UniqueId<SpecificationGoal>, Map<UniqueId<InstanceGoal>, InstanceGoal>> instanceGoalsBySpecificationGoal = new ConcurrentHashMap<>();
    final Map<UniqueId<Pmf>, Pmf> pmfs = new ConcurrentHashMap<>();
    final Map<UniqueId<Policy>, Policy> policies = new ConcurrentHashMap<>();
    final Map<UniqueId<Role>, Role> roles = new ConcurrentHashMap<>();
    final Map<UniqueId<SpecificationGoal>, SpecificationGoal> specificationGoals = new ConcurrentHashMap<>();

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/Entities.kt
package runtimemodels.chazm.model

import runtimemodels.chazm.api.entity.Policy
import runtimemodels.chazm.api.entity.Role
import runtimemodels.chazm.api.entity.SpecificationGoal
import runtimemodels.chazm.api.id.PolicyId
import runtimemodels.chazm.api.id.RoleId
import runtimemodels.chazm.api.id.SpecificationGoalId
import java.util.concurrent.ConcurrentHashMap

internal class Entities {

    //    val agents: MutableMap<AgentId, Agent> = ConcurrentHashMap()
//    val attributes: MutableMap<AttributeId, Attribute> = ConcurrentHashMap()
//    val capabilities: MutableMap<CapabilityId, Capability> = ConcurrentHashMap()
//    val characteristics: MutableMap<CharacteristicId, Characteristic> = ConcurrentHashMap()
//    val instanceGoals: MutableMap<InstanceGoalId, InstanceGoal> = ConcurrentHashMap()
//    val instanceGoalsBySpecificationGoal: MutableMap<SpecificationGoalId, MutableMap<InstanceGoalId, InstanceGoal>> = ConcurrentHashMap()
//    val pmfs: MutableMap<PmfId, Pmf> = ConcurrentHashMap()
    val policies: MutableMap<PolicyId, Policy> = ConcurrentHashMap()
    val roles: MutableMap<RoleId, Role> = ConcurrentHashMap()
    val specificationGoals: MutableMap<SpecificationGoalId, SpecificationGoal> = ConcurrentHashMap()

}